src/diffusion_for_multi_scale_molecular_dynamics/active_learning_loop/active_learning.py

```diff
@@ -0,0 +1,322 @@
+                mapped_uncertainty_file_path=mapped_uncertainty_file_path,
+            )
+            logger.info(f"  ARTn state is {calculation_state}")
+
+            if calculation_state == CalculationState.SUCCESS:
+                logger.info("Active Learning Campaign is Complete.")
+
+                logger.info("Writing FLARE model checkpoint.")
+                checkpoint_path = working_directory / "trained_flare.json"
+                flare_trainer.write_checkpoint_to_disk(checkpoint_path)
+                logger.info("Exiting.")
+                break
+
+            logger.info("  Extracting uncertain structure from ARTn work directory...")
+            uncertain_structure, uncertainty_per_atom = (
+                self._get_uncertain_structure_and_uncertainties(artn_working_directory)
+            )
+
+            number_of_uncertain_envs = np.sum(
+                uncertainty_per_atom > uncertainty_threshold
+            )
+            logger.info(
+                f" -> There are {number_of_uncertain_envs} environments with uncertainty above the threshold."
+            )
+
+            logger.info("  Making new samples based on uncertainties.")
+            list_sample_structures, list_sample_information = self._make_samples(
+                uncertain_structure, uncertainty_per_atom
+            )
+
+            logger.info("  Labelling samples with oracle...")
+            time1 = time.time()
+            list_single_point_calculations = [
+                self.oracle_single_point_calculator.calculate(structure)
+                for structure in list_sample_structures
+            ]
+            time2 = time.time()
+            logger.info(
+                f" -> It took {time2- time1: 6.2e} seconds to compute labels with Oracle."
+            )
+
+            logger.info("  Converting labelled samples and writing pickle to disk.")
+            oracle_df = self._convert_single_point_calculations_to_dataframe(
+                list_single_point_calculations, list_sample_information
+            )
+
+            oracle_directory = current_sub_directory / "oracle"
+            oracle_directory.mkdir(parents=True, exist_ok=True)
+
+            output_file = oracle_directory / "oracle_single_point_calculations.pkl"
+            oracle_df.to_pickle(output_file)
+
+            logger.info("  Adding samples and uncertain environment to FLARE.")
+            for single_point_calculation, sample_info in zip(list_single_point_calculations, list_sample_information):
+                active_environment_indices = self._determine_active_environment_indices(sample_info,
+                                                                                        uncertainty_threshold)
+                flare_trainer.add_labelled_structure(
+                    single_point_calculation,
+                    active_environment_indices=active_environment_indices,
+                )
+
+            logger.info("  Fitting the FLARE hyperparameters...")
```

Comment:
note that we should make this logging more agnostic to the method. Maybe not in this PR, but something to keep in mind as a TODO